### PR TITLE
Cleaned up the SDK (List, ArrayList, HashMap) a bit

### PR DESCRIPTION
**Pull request by marcusnaslund**
Moved the contents of `lang/equalities` back to `HashMap` since `ArrayList` won't need them anymore.

### CODE REVIEW

File: source/sdk/structs/HashMap.ooc [hunk 1]
```diff
@@ -137,8 +163,6 @@ HashMap: class <K, V> extends BackIterable<V> {
 
 		keyEquals = getStandardEquals(K)
 		hashKey = getStandardHashFunc(K)
-
-		T = V // workarounds ftw
```
Comment (marcusnaslund) [inline source/sdk/structs/HashMap.ooc]: I honestly don't know what this was supposed to do.

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/sdk/structs/ArrayList.ooc [hunk 2]
```diff
@@ -75,25 +75,6 @@ ArrayList: class <T> extends List<T> {
...
-			if (this as List equals(this[index], element)) return index
-			index += 1
-		}
-		return -1
-	}
-
-	lastIndexOf: override func (element: T) -> SSizeT {
-		index := _size
-		while (index > -1) {
-			if (memcmp(data + index * T size, element, T size) == 0) return index
```
Comment (sebastianbaginski) [inline source/sdk/structs/ArrayList.ooc]: What a broken function, they start from incorrect index and check for a condition that can never be true [in `while`, -1 is converted to a very big unsigned number] :)

File: source/sdk/structs/HashMap.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/sdk/structs/HashMap.ooc]: Maybe a hack for a very old rock version ?

Comment (sebastianbaginski) [inline source/sdk/structs/HashMap.ooc]: Good that it is gone now anyway :+1:

File: source/sdk/structs/ArrayList.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/sdk/structs/ArrayList.ooc]: This is what happens when there are no unit tests :+1:

**Comment by marcusnaslund** [comment]
@jonathanudd Make sure you rebase your work on `HashMap`/`HashDictionary` now.

**Comment by jonathanudd** [comment]
I will, thanks.

**Pull request merged**